File: sentimental-credit/credit.py
def getInt():  # Function to check if input is a number or not
    try:
        num = int(input("Number: "))
        return num
    except:
        getInt()


num = getInt()
sum = 0
strNum = str(num)
i = 0
for x in range(len(strNum)-1, -1, -1):  # Calculating checksum
    if i % 2 == 0:
        sum += int(strNum[x])
    else:
        c = strNum[x]
        a = int(c) * 2
        b = str(a)
        if len(b) > 1:
            sum += int(b[0])
            sum += int(b[1])
        else:
            sum += a
    i += 1
if sum % 10 == 0:  # If sum has a 0 as its last digit, number has passed checksum
    a = (str(strNum))[0:2]
    lenNum = len(str(strNum))
    if lenNum == 15:
        if a in ["34", "37"]:
            print("AMEX")
    elif lenNum == 16:
        if a in ["51", "52", "53", "54", "55"]:
            print("MASTERCARD")
        elif (str(strNum))[0] == "4":
            print("VISA")
    elif lenNum == 13:
        if (str(strNum))[0] == "4":
            print("VISA")
    else:
        print("INVA2LID")
else:
    print("INVALID")

File: credit/credit.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

// Luhn’s Algorithm
bool checksum(long num, int n)
{
    if (n == 0)
    {
        return false;
    }
    int sum = 0;
    for (int i = 2; i <= n; i += 2)
    {
        long a = 2 * ((num % (long)pow(10, i)) / (long)pow(10, i - 1));
        int b = (int)a;
        if (b > 9)
        {
            sum += b % 10 + (b % 100) / 10;
        }
        else
        {
            sum += b;
        }
    }
    for (int i = 1; i <= n; i += 2)
    {
        long a = (num % (long)pow(10, i)) / (long)pow(10, i - 1);
        sum += (int)a;
    }
    if ((sum % (int)pow(10, 1) == 0))
    {
        return true;
    }
    else
    {
        return false;
    }
}
int main(void)
{
    long num;
    int n;
    num = get_long("Number: ");

    // Calculating number of digits, if
    if (num < 1000000000000  || (num > 9999999999999 && num < 100000000000000) || num > 9999999999999999)
    {
        n = 0;
    }
    else if (num > 999999999999 && num < 10000000000000)
    {
        n = 13;
    }
    else if (num > 99999999999999 && num < 1000000000000000)
    {
        n = 15;
    }
    else
    {
        n = 16;
    }
    if (checksum(num, n))
        // if credit card number is valid, check for which credit card is it.
    {
        switch (n)
        {
            case 13:
                if (num >= 4000000000000 && num < 5000000000000)
                {
                    printf("VISA\n");
                }
                else
                {
                    printf("INVALID\n");
                }
                break;

            case 15:
                if ((num >= 340000000000000 && num < 350000000000000) || (num >= 370000000000000 && num < 380000000000000))
                {
                    printf("AMEX\n");
                }
                else
                {
                    printf("INVALID\n");
                }
                break;

            case 16:
                if (num >= 5100000000000000 && num < 5600000000000000)
                {
                    printf("MASTERCARD\n");
                }
                else if (num >= 4000000000000000 && num < 5000000000000000)
                {
                    printf("VISA\n");
                }
                else
                {
                    printf("INVALID\n");
                }
                break;
        }
    }
    else
    {
        printf("INVALID\n");
    }
}